### DISCUSSION TITLE
[Q&A] shipping_profile_id and some ID fields have values are larger than allowed for INTEGER.

### DISCUSSION THREAD

**Discussion by duongvd2106** [etsy/open-api#1001] [Q&A] [answered]
Hi, I am integrating API v3 into my Java web application. And I tried creating a feature based on the createDraftListing endpoint.

I found that some shipping_profile_id values in my test store are greater than the maximum limit for an integer (in Java, the maximum integer value is 2147483647, but the shipping_profile_id I got from API v3 value is up to 185579070499). Therefore, I cannot include it in the createDraftListing request to POST it to Etsy.

Although the reference (https://developers.etsy.com/documentation/reference#operation/createDraftListing) describes this field as an integer type, and not only shipping_profile_id but almost all ID fields are described with a value type of integer. 
However, I have found that there are many ID values, such as receipt_id and transaction_id..,  that exceed the integer's maximum threshold.

Am I doing something wrong? What can I do to POST a draft listing with shipping_profile_id great than the maximum integer value?

Thanks for reading.

**Comment by duongvd2106** [comment, marked as answer]
Looks like I misunderstood the API documentation. The Integer mentioned there implies a positive integer and the maximum value is not the maximum value of a 32-bit signed integer like I thought. I passed the data type from Java Integer to Java Long and everything seems to be fine.

Thanks.

**Reply by Gareth064** [reply]
Yeah. This is an annoying documentation flaw that Etsy have refused to fix in the past.

Anything that denotes an ID, is a long or BigInt (language dependant)
